src/java/org/apache/cassandra/metrics/MicrometerBufferPoolMetrics.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.metrics;
+
+import com.codahale.metrics.Meter;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tags;
+import org.apache.cassandra.utils.memory.BufferPool;
+
+import static org.apache.cassandra.metrics.CassandraMetricsRegistry.Metrics;
+
+public class MicrometerBufferPoolMetrics extends MicrometerMetrics implements BufferPoolMetrics
+{
+    private static final String METRICS_PREFIX = "buffer_pool";
+
+    public static final String TOTAL_SIZE_BYTES = METRICS_PREFIX + "_total_size_bytes";
+    public static final String USED_SIZE_BYTES = METRICS_PREFIX + "_used_size_bytes";
+    public static final String OVERFLOW_SIZE_BYTES = METRICS_PREFIX + "_overflow_size_bytes";
+    public static final String OVERFLOW_ALLOCATIONS = METRICS_PREFIX + "_overflow_allocations";
+    public static final String NAME_TAG = "pool_name";
+
+    private final String scope;
+    private final BufferPool bufferPool;
+
+    /** Total number of hits */
+    private final Meter hits;
+
+    /** Total number of misses */
+    private final Meter misses;
+
+    public MicrometerBufferPoolMetrics(String scope, BufferPool bufferPool)
+    {
+        super();
+
+        this.scope = scope;
+        this.bufferPool = bufferPool;
+        MetricNameFactory factory = new DefaultNameFactory("BufferPool", scope);
+        this.hits = Metrics.meter(factory.createMetricName("Hits"));
```

Comment:
Shouldn't we declare `hits` and `misses` the same way as similar Micrometer metrics, that is:
1) Configure the name as a static final class attribute.
2) Use `counter()`.

Reply:
For context, I'll try and explain how I arrived at this - happy to make changes as necessary:

To start, afaict the `overflowAllocs` "missed allocations" in DSE are the same as plain "misses" in C*, so I kept the "misses" (and "hits") terminology usage up till here in `MicrometerBufferPoolMetrics` - since CNDB already expects the metric name to be `OVERFLOW_ALLOCATIONS` the `register` method creates the gauge named `OVERFLOW_ALLOCATIONS` backed by the `misses` meter using `getMeanRate`.

`misses` is a meter here because that seemed to me to be the closest thing to the DSE line:
```
this.overflowAllocs = Meter.makeSingle();
```

So ... `OVERFLOW_ALLOCATIONS` static name backed by a meter is the same here as in DSE, I think; just that the name of the backing meter here is `misses` in an attempt to remain consistent with the interface and C* usage.

I was not exactly sure how to handle `hits` - used in C*, but not in DSE afaict - so there is a `hits` metric here, but didn't register a MicrometerMetrics gauge since CNDB isn't expecting one, and I wasn't quite sure how to name it ("NORMAL_ALLOCATIONS"?).

Re: using `counter()`
Both here and in DSE the micrometer gauge uses `Meter::getMeanRate` so it has to be backed by a `Meter` (or perhaps I misunderstood).

Reply:
> since CNDB already expects the metric name to be OVERFLOW_ALLOCATIONS the register method creates the gauge named OVERFLOW_ALLOCATIONS backed by the misses meter using getMeanRate.

Ok I see why you made `misses` like that: IIUIC you didn't plan to expose it by itself, but only via `OVERFLOW_ALLOCATIONS`. That makes sense, but then I would not create a metric, but simply declare it the same way we declare `overflowAllocs` in CNDB:

`this.misses = Meter.makeSingle();`

Re: `hits`, let's do the same and expose it as a meter-backed `gauge` named "pool allocations"?

Re: using `counter`, ignore that, I just misunderstood why you exposed hits/misses, thanks again for clarifying.

Reply:
Sounds good, I'll add `POOL_ALLOCATIONS` as a meter backed gauge as suggested.

I see what you mean about not creating a (registered) metric with `Metrics.meter`, and since `Meter.makeSingle` doesn't currently exist in C*/CC, probably we can simply use `new Meter()` instead.

Reply:
@sbtourist wdyt of the latest update?